src/crimson/os/seastore/cached_extent.h

```diff
@@ -385,7 +385,7 @@ class CachedExtent
    * apply_delta override for LogicalCachedExtent implementers.
    */
   virtual void apply_delta_and_adjust_crc(
-    paddr_t base, const ceph::bufferlist &bl) = 0;
+    paddr_t base, const ceph::bufferlist &bl, bool checksum_offloaded_to_device) = 0;
```

Comment:
I think the device protection should only offload crcs for the extents directly stored in the disk, it isn't related to the other cases.

So I think here crc is still needed to validate that delta applying keeps reasonable, unexpected software changes may result in incompatible replaying, which should be detected and it is unrelated to disk offloading.

In another word, seems to me crcs are still always needed for delta-based mutations.